### DISCUSSION TITLE
Bias Parameters!!

### DISCUSSION THREAD

**Discussion by Brayan532** [ultralytics/ultralytics#16218] [Q&A]
Hi there,
I’ve encountered an unexpected behavior when adding a simple CNN layer with 512 input and output channels, stride=1, and bias enabled (without using the Conv class defined in YOLO). Instead of the expected 512 bias parameters, only 128 parameters are added. Could you help explain this observation? simply put, 
With bias enabled, N is the total number of parameters
With bias=False, N-128 is the total number of parameters.

**Comment by glenn-jocher** [comment]
@Brayan532 it sounds like the issue might be related to how the bias parameters are being initialized or counted in your setup. Ensure that the layer's configuration aligns with your expectations, and check if any custom settings or framework defaults might be affecting the parameter count. If you're using a specific library, reviewing its documentation on layer initialization could provide further insights.
